File: truss-compiler/src/main/java/truss/compiler/preprocessor/DefineDirective.java
package truss.compiler.preprocessor;

import org.apache.commons.lang.Validate;

class DefineDirective implements Directive {
    private final String identifier;
    private final boolean define;

    public DefineDirective(String identifier, boolean define) {
        Validate.notNull(identifier, "identifier");

        this.identifier = identifier;
        this.define = define;
    }

    public String getIdentifier() {
        return identifier;
    }

    public boolean isDefine() {
        return define;
    }

    @Override
    public DirectiveKind getKind() {
        return DirectiveKind.DEFINE;
    }
}

File: Truss.Compiler/PreProcessor/DefineDirective.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.PreProcessor {
    internal class DefineDirective : IDirective {
        public DefineDirective(string identifier, bool define) {
            if (identifier == null) {
                throw new ArgumentNullException("identifier");
            }

            Identifier = identifier;
            IsDefine = define;
        }

        public string Identifier { get; private set; }

        public bool IsDefine { get; private set; }

        public DirectiveKind Kind {
            get { return DirectiveKind.Define; }
        }
    }
}
